gmso/external/convert_dataframe.py

```diff
@@ -0,0 +1,381 @@
+        'specific_columns' will only output the attributes from the `columns` argument.
+        'publication' will use the default outputs, but remove duplicate values from the dataframes. It adds a column labeled
+        'Atom Indices' to the `sites` dataframe, which enumerates the indices that the atom_type is a part of.
+        `remove_duplicates` will remove duplicate rows from the dataframe. For sites, this column is `atom_types.name`.
+        For connections, it is the `connection_types.connection_members`. For sites, an additional column will be added, labeled
+        `Atom Indices` that includes the site indexes of members that are the given `atom_type.name`. Because these methods
+        are specific to a given Topology element, the `parameters` argument must be one of
+        {"sites", "bonds", "angles", "dihedrals", "impropers"}, not {"all"}.
+    columns : list of str, optional, default=None
+        List of strings that are attributes of the topology site and can be included as entries in the pandas dataframe.
+        Examples of these can be found by printing `topology.sites[0].__dict__` or `topology.bonds[0].__dict__`.
+        See https://gmso.mosdef.org/en/stable/data_structures.html#gmso.Atom for additional information on labeling.
+    handle_unyts: str, optional, default='to_headers'
+        The placement/recording of unyt quantities in dataframe.
+        Options are 'to_headers', 'with_data', 'no_unyts'
+        Determines if numerical values in the DataFrame are saved as unyt quantities or floats. Default case, 'to_headers",
+        puts the unyts as strings to go with the column header of the dataframe.
+        `with_data` leaves any values alone, so any values in the Topology that are unyt quantities will stay that way.
+        `no_unyts` strips any unyt values and converts to a float in the associated element of the dataframe.
+        See https://unyt.readthedocs.io/en/stable/usage.html
+        for more information about manipulating unyt quantities.
+
+    Returns
+    -------
+    Dictionary of Pandas Dataframe
+        A python dictionary of pandas.Dataframe object, see https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.html
+        for further information. The keys of this dictionary are the attributes of the topology that are associated with each DataFrame. These
+        can be `sites`, `bonds`, `angles`, `dihedrals`, `impropers`, which are determined from the argument `parameters`.
+
+
+    Examples
+    --------
+    # example topology to use
+    ``` python
+    >>> import gmso
+    >>> import mbuild as mb
+    >>> from gmso.parameterization import apply
+    >>> cpd = mb.load("C", smiles=True)
+    >>> top = cpd.to_gmso()
+    >>> ff = gmso.ForceField("oplsaa")
+    >>> ptop = apply(top, ff)
+    ```
+
+
+    >>> gmso.external.convert_dataframe.to_dataframeDict(ptop, parameters='sites', columns=['charge'], handle_unyts="to_headers")
```

Comment:
@Zeerakkhan47 I think you could expand the information passed into the `columns` parameter here to include some of the things @CalCraven mention in his comment on June 16th.